packages/filecoin-actor-utils/src/utils/code.ts

```diff
@@ -1,44 +1,53 @@
+import base32Decode from 'base32-decode'
 import { networkActorCodeMap, networkActorCodeMapInv } from '../data'
-import { ActorCode, ActorName, NetworkName } from '../types'
+import { ActorCode, ActorName, LotusCID, NetworkName } from '../types'
 
 /**
- * Resolves the actor name by providing the actor code. When also providing
- * the network name, it will limit the search to that specific network.
+ * Resolves the actor name by providing the actor code and the network name.
  * @param actorCode the actor code which is used to resolve the actor name
- * @param networkName (optional) the network in which to search for the actor name
+ * @param networkName the network in which to search for the actor name
  * @returns the actor name when found or null when not found
  */
 export const getActorName = (
-  actorCode: ActorCode,
-  networkName?: NetworkName
+  actorCode: ActorCode | LotusCID,
+  networkName: NetworkName
 ): ActorName | null => {
-  // If "networkName" is provided, return the actor name
-  // for that specific network or null when not found
-  if (networkName) {
-    return networkActorCodeMapInv[networkName]?.[actorCode] ?? null
-  }
+  const code = typeof actorCode === 'string' ? actorCode : actorCode['/']
```

Comment:
Typescript would probably catch this, but if we ever didn't pass a CID, or a CID was `null` that accidentally got passed through to this function, then `actorCode['/']` would result in an unhandled runtime error. Probably safe to just `actorCode?.['/']`

Reply:
Good call